compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1492,7 +1492,7 @@ trait Applications extends Compatibility {
 
         val dummyArg = dummyTreeOfType(ownType)
         val (newUnapplyFn, unapplyApp) =
-          val unapplyAppCall = withMode(Mode.NoInline):
+          val unapplyAppCall = withMode(Mode.NoInlineUnapply):
```

Comment:
Would be good to add an explanation why unapplys should not be inlined here.

Reply:
We are running out of mode bits. Is there some other way to know this info? For instance we could observe that the `unapply` is applied to a dum,yArg and not inline in this case?

Reply:
Now the implementation is based on the existence of the dummy. No extra mode required.